libs/utils/wa_results_collector.py

```diff
@@ -292,14 +292,26 @@ def _read_wa_dir(self, wa_dir):
                 # If the workload spec has a 'test' classifier, use that to
                 # identify it.
                 test = classifiers.pop('test')
+
+                # Use the workload name if not set; see comment in else branch
+                if not test:
+                    test = workload
             elif 'test' in job['workload_parameters']:
                 # If not, some workloads have a 'test' workload_parameter, try
                 # using that
                 test = job['workload_parameters']['test']
+
+                # Use the workload name if not set; see comment in else branch
+                if not test:
+                    test = workload
             elif 'test_ids' in job['workload_parameters']:
                 # If not, some workloads have a 'test_ids' workload_parameter, try
                 # using that
                 test = job['workload_parameters']['test_ids']
+
+                # Use the workload name if not set; see comment in else branch
+                if not test:
+                    test = workload
```

Comment:
Sorry for the stall on this one, I unconsciously tend to avoid wltests PRs since I'm not very fluent in it.

Could we clean the handling a bit?

```python
# Use the workload name as a default fallback.
# This isn't ideal because it means the results from jobs with
# different workload parameters will be amalgamated.
test = workload

if 'test' in classifiers:
    test = classifiers.pop('test') or test
elif x in y:
    test = job[x] or test
[...]
```

Reply:
This is less readable for me - but it could be only me.

will classifiers.pop() remove the item from the list too or just read it?

Reply:
It removes the item. I don't know why it's done here, I was just commenting on the look of the change.

This was mostly about avoiding a
```python
if not test:
    test = workload
```
in every single condition.


If you don't like the or'ing, you could go with something like this:

```python
test = None

if 'test' in classifiers:
    test = foo()
elif x in y:
    test = bar()
[...]

if not test:
    # blurb about workload name here
    test = workload
```

Reply:
Ah OK I get you now. So your suggestion to basically append `or workload` whenever we assign test in the if else legs? Then yeah that would be better.